aws/resource_aws_sagemaker_model_test.go

```diff
@@ -0,0 +1,576 @@
+package aws
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/sagemaker"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"log"
+	reflect "reflect"
+	"strings"
+	"testing"
+)
+
+func init() {
+	resource.AddTestSweepers("aws_sagemaker_model", &resource.Sweeper{
+		Name: "aws_sagemaker_model",
+		F:    testSweepSagemakerModels,
+	})
+}
+
+func testSweepSagemakerModels(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).sagemakerconn
+
+	req := &sagemaker.ListModelsInput{
+		NameContains: aws.String("terraform-testacc-sagemaker-model"),
+	}
+	resp, err := conn.ListModels(req)
+	if err != nil {
+		return fmt.Errorf("Error listing models: %s", err)
+	}
+
+	if len(resp.Models) == 0 {
+		log.Print("[DEBUG] No sagemaker models to sweep")
+		return nil
+	}
+
+	for _, model := range resp.Models {
+		_, err := conn.DeleteModel(&sagemaker.DeleteModelInput{
+			ModelName: model.ModelName,
+		})
+		if err != nil {
+			return fmt.Errorf(
+				"Error deleting sagemaker model (%s): %s",
+				*model.ModelName, err)
+		}
+	}
+
+	return nil
+}
+
+func TestAccAWSSagemakerModel_basic(t *testing.T) {
+	var model sagemaker.DescribeModelOutput
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckSagemakerModelDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSagemakerModelConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSagemakerModelExists("aws_sagemaker_model.foo", &model),
+					testAccCheckSagemakerModelName(&model, "terraform-testacc-sagemaker-model-foo"),
+					testAccCheckSagemakerModelExecutionRoleArn(&model, "terraform-testacc-sagemaker-model-foo"),
+					testAccCheckSagemakerModelPrimaryContainerImage(&model,
+						"174872318107.dkr.ecr.us-west-2.amazonaws.com/kmeans:1"),
+
+					resource.TestCheckResourceAttr(
+						"aws_sagemaker_model.foo", "name", "terraform-testacc-sagemaker-model-foo"),
+					resource.TestCheckResourceAttr("aws_sagemaker_model.foo", "primary_container.#", "1"),
+					resource.TestCheckResourceAttr(
+						"aws_sagemaker_model.foo", "primary_container.0.image",
+						"174872318107.dkr.ecr.us-west-2.amazonaws.com/kmeans:1"),
+					resource.TestCheckResourceAttrSet("aws_sagemaker_model.foo", "execution_role_arn"),
+					resource.TestCheckResourceAttrSet("aws_sagemaker_model.foo", "arn"),
+					resource.TestCheckResourceAttrSet("aws_sagemaker_model.foo", "creation_time"),
+				),
+			},
```

Comment:
It doesn't look like there are any checks for import support. We can add 
```
{
	ResourceName:      resourceName,
	ImportState:       true,
	ImportStateVerify: true,
},
```
to make sure we can import properly